### PR TITLE
Add cifar 10 SGD

### PR DESCRIPTION
**Pull request by ByzanTine**

### CODE REVIEW

File: cifar10/cifar10.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+from six.moves import urllib
+import tensorflow as tf
+
+import cifar10_input
+
+FLAGS = tf.app.flags.FLAGS
+
+# Basic model parameters.
+tf.app.flags.DEFINE_integer('batch_size', 2500,
```
Comment (ByzanTine) [inline cifar10/cifar10.py]: This is batch size.

File: cifar10/cifar10.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+  Returns:
+    Logits.
+  """
+    # We instantiate all variables using tf.get_variable() instead of
+    # tf.Variable() in order to share variables across multiple GPU training runs.
+    # If we only ran this model on a single GPU, we could simplify this function
+    # by replacing all instances of tf.get_variable() with tf.Variable().
+    #
+    # conv1
+    flat_image = tf.contrib.layers.flatten(images)
```
Comment (ByzanTine) [inline cifar10/cifar10.py]: Here is the network architecture

File: cifar10/cifar10.py [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    decay_steps = int(num_batches_per_epoch * NUM_EPOCHS_PER_DECAY)
+    lr = tf.train.exponential_decay(
+        INITIAL_LEARNING_RATE,
+        global_step,
+        decay_steps,
+        LEARNING_RATE_DECAY_FACTOR,
+        staircase=True)
+
+    tf.summary.scalar('learning_rate', 0.05)
+    optimizer = tf.train.MomentumOptimizer(0.05, 0.9)
```
Comment (ByzanTine) [inline cifar10/cifar10.py]: This is the optimizer.

Train method builds a train operator that applies to all the trainable variables.

File: cifar10/cifar10.py (hunk 2 above)
Comment (montaguegabe) [reply cifar10/cifar10.py]: Is this the same as the architecture used in MATLAB?

Comment (ByzanTine) [reply cifar10/cifar10.py]: I think so, they use this sort of 3 layer network with 1 hidden layer and 1 output layer as softmax.

**Comment by ByzanTine** [comment]
Though, I still don't have my SGD converge in 10^6 runs (each run has batch size 1000)...
And 10^6 is taken already 3hours...
